src/popover/popover.jsx

```diff
@@ -0,0 +1,355 @@
+    return {
+      open: false,
+    };
+  },
+
+  contextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+
+  windowListeners: {
+    resize: 'setPlacementThrottled',
+    scroll: 'setPlacementThrottled',
+  },
+
+  componentWillReceiveProps(nextProps) {
+    if (this.state.openedManually)
+      return;
+
+    if (nextProps.open !== this.state.open) {
+      if (nextProps.open)
+        this._showInternal(nextProps.anchorEl);
+      else
+        this.hide();
+    }
+  },
+
+  componentDidUpdate() {
+    this.setPlacement();
+  },
+
+  render() {
+    return <RenderToLayer
+      ref="layer"
+      {...this.props}
+      componentClickAway={this.componentClickAway}
+      render={this.renderLayer} />;
+  },
+
+  renderLayer() {
+    let {
+      animated,
+      targetOrigin,
+      className,
+      zDepth,
+    } = this.props;
+
+    const anchorEl = this.props.anchorEl || this.anchorEl;
+    let anchor = this.getAnchorPosition(anchorEl);
+    let horizontal = targetOrigin.horizontal.replace("middle", "vertical");
+
+    let wrapperStyle = {
+      position: 'fixed',
+      top: anchor.top, left:
+      anchor.left,
+      zIndex: 20,
+      opacity:1,
+      overflow:'auto',
+      maxHeight:'100%',
+      transform:'scale(0,0)',
+      transformOrigin: `${horizontal} ${targetOrigin.vertical}`,
+      transition: animated ? Transitions.easeOut('500ms', ['transform', 'opacity']) : null,
+    };
+    wrapperStyle = this.mergeAndPrefix(wrapperStyle, this.props.style);
+
+    let horizontalAnimation = {
+      maxHeight:'100%',
+      overflowY:'auto',
+      transform:'scaleX(0)',
+      opacity:1,
+      transition: animated ? Transitions.easeOut('250ms', ['transform', 'opacity']): null,
+      transformOrigin: `${horizontal} ${targetOrigin.vertical}`,
+    };
+
+    let verticalAnimation = {
+      opacity:1,
+      transform:'scaleY(0)',
+      transformOrigin: `${horizontal} ${targetOrigin.vertical}`,
+      transition: animated ? Transitions.easeOut('500ms', ['transform', 'opacity']) : null,
+    }
+
+    const open = this.state.open;
+
+    return (
+      <Paper style={wrapperStyle} zDepth={zDepth} className={className} >
+        <div>
+          <div style={horizontalAnimation}>
+            <div style={verticalAnimation}>
+                {this.props.children}
+           </div>
+          </div>
+        </div>
+      </Paper>
+    );
+  },
+
+  requestClose() {
+    if (this.props.onRequestClose)
+      this.props.onRequestClose();
+  },
+
+  componentClickAway(e) {
+    if (e.defaultPrevented) {
+      return;
+    }
+    this.hide();
+  },
+
+  _resizeAutoPosition() {
+    this.setPlacement();
+  },
+
+  _showInternal(anchorEl) {
+    this.anchorEl = anchorEl || this.props.anchorEl;
+    this.setState({open: true});
+    const popOverShowEvent = new CustomEvent('popOverOnShow', {detail: this});
+    document.dispatchEvent(popOverShowEvent);
+
+  },
+
+  show(anchorEl) {
```

Comment:
Do we need those to imperative `show` and `hide` method? I would rather get ride of them if we can use the `open` property to achieve the same behavior.